### PR TITLE
Optimize Run + Design Except Feature + Various

### PR DESCRIPTION
**Pull request by nicolas-kuechler**
# Summary

## Optimize config/dir creation
As part of the setup, before running experiments, we create all working directories and place the `config.json` in the folder. 
Until now, this relied on pure Ansible. However, for many jobs, this creation becomes a bottleneck. 
Now, there is a custom module that does this more efficiently.

## Optimize result fetching
We replaced the slow fetching of results with an additional custom module.
More efficiency is possible because now `tsp` or `slurm` do not need to process a completing job id one-by-one.
Instead, they can report that a list of job ids finished and can be downloaded.

## New Experiment Design Feature: `except_filters`
We added a basic implementation of #71 that allows filtering out certain combinations.

## New Super ETL Feature: `pipelines` filter 
We add the possibility when running  a super_etl via make to run only a subset of pipelines (see `pipelines="a b"`)

## ETL Extract Optimization
We only flatten results that require flattening, which speeds up the processing.

## ETL Step
We add a new ETL Loader that allows storing a data frame as a pickle.

## Option to save ETL results to Notion
We add a utility function that allows storing the results of a loader to a Notion page.

### CODE REVIEW

File: ansible.cfg [hunk 1]
```diff
@@ -13,11 +13,13 @@ any_errors_fatal_setup = true
...
 stdout_callback = community.general.selective
 
+
 # speedup by using ssh pipelining
 pipelining = True
-[ssh_connection]
-pipelining = True
-ssh_args = -o ControlMaster=auto -o ControlPersist=1200
\ No newline at end of file
+#[ssh_connection]
```
Comment (hiddely) [inline ansible.cfg:23]: Is this disabled on purpose?

File: doespy/doespy/etl/etl_base.py [hunk 2]
```diff
@@ -474,7 +497,9 @@ def extract(
         runs = _list_dir_only(exp_dir)
         factor_columns = _parse_factors(base_experiments[exp])
 
-        for run in runs:
+        from tqdm import tqdm
```
Comment (hiddely) [inline doespy/doespy/etl/etl_base.py:500]: maybe move this import to top of file?

File: doespy/doespy/etl/etl_base.py [hunk 3]
```diff
@@ -616,4 +643,7 @@ def _is_file(path, f):
 
 
 def _flatten_d(d):
-    return json.loads(pd.json_normalize(d, sep=".").iloc[0].to_json())
+    if any(isinstance(i, dict) for i in d.values()):
```
Comment (hiddely) [inline doespy/doespy/etl/etl_base.py:646]: ✨

**Review by hiddely — COMMENTED** [review]
Looks good apart from two minor issues 😊

File: ansible.cfg (hunk 1 above)
Comment (nicolas-kuechler) [reply ansible.cfg]: I don't remember why I commented this out.
I'm bringing it back.

**Pull request merged**